test/e2e/e2e_test.go

```diff
@@ -240,24 +209,145 @@ func TestLowNodeUtilization(t *testing.T) {
+								{
+									MatchFields: []v1.NodeSelectorRequirement{
+										{Key: "metadata.name", Operator: v1.NodeSelectorOpIn, Values: []string{workerNodes[0].Name}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		t.Logf("Creating pod %v in %v namespace for node %v", pod.Name, pod.Namespace, workerNodes[0].Name)
+		_, err := clientSet.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("Error creating LNU pods: %v", err)
+			if err = clientSet.CoreV1().Pods(pod.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"test": "node-utilization", "name": "test-rc-node-utilization"})).String(),
+			}); err != nil {
+				t.Fatalf("Unable to delete LNU pods: %v", err)
+			}
+			return
+		}
+	}
+
+	t.Log("Creating RC with 4 replicas owning the created pods")
+	rc := RcByNameContainer("test-rc-node-utilization", testNamespace.Name, int32(4), map[string]string{"test": "node-utilization"}, nil, "")
 	if _, err := clientSet.CoreV1().ReplicationControllers(rc.Namespace).Create(ctx, rc, metav1.CreateOptions{}); err != nil {
-		t.Errorf("Error creating deployment %v", err)
+		t.Fatalf("Error creating RC %v", err)
 	}
+	defer deleteRC(ctx, t, clientSet, rc)
+	waitForRCPodsRunning(ctx, t, clientSet, rc)
 
-	evictPods(ctx, t, clientSet, nodeInformer, nodes, rc)
-	deleteRC(ctx, t, clientSet, rc)
+	// Run LowNodeUtilization strategy
+	evictionPolicyGroupVersion, err := eutils.SupportEviction(clientSet)
+	if err != nil || len(evictionPolicyGroupVersion) == 0 {
+		t.Fatalf("%v", err)
+	}
+	podEvictor := evictions.NewPodEvictor(
+		clientSet,
+		evictionPolicyGroupVersion,
+		false,
+		0,
+		nodes,
+		true,
+		false,
+		false,
+	)
+
+	podsOnMosttUtilizedNode, err := podutil.ListPodsOnANode(ctx, clientSet, workerNodes[0], podutil.WithFilter(podEvictor.Evictable().IsEvictable))
+	if err != nil {
+		t.Errorf("Error listing pods on a node %v", err)
+	}
+	podsBefore := len(podsOnMosttUtilizedNode)
+
+	t.Log("Running LowNodeUtilization strategy")
+	strategies.LowNodeUtilization(
+		ctx,
+		clientSet,
+		deschedulerapi.DeschedulerStrategy{
+			Enabled: true,
+			Params: &deschedulerapi.StrategyParameters{
+				NodeResourceUtilizationThresholds: &deschedulerapi.NodeResourceUtilizationThresholds{
+					Thresholds: deschedulerapi.ResourceThresholds{
+						v1.ResourceCPU: 70,
+					},
+					TargetThresholds: deschedulerapi.ResourceThresholds{
+						v1.ResourceCPU: 80,
+					},
+				},
+			},
+		},
+		workerNodes,
+		podEvictor,
+	)
+
+	podsOnMosttUtilizedNode, err = podutil.ListPodsOnANode(ctx, clientSet, workerNodes[0], podutil.WithFilter(podEvictor.Evictable().IsEvictable))
+	if err != nil {
+		t.Errorf("Error listing pods on a node %v", err)
+	}
+	podsAfter := len(podsOnMosttUtilizedNode)
+
+	if podsBefore > podsAfter {
+		t.Fatalf("No pod has been evicted from %v node", workerNodes[0].Name)
+	}
```

Comment:
I think `podsAfter` is supposed to be less than `podsBefore` given some pods will be evicted from `MosttUtilizedNode` by `LNU`.

Reply:
Can you add some logs to show the actual value of `podsBefore` and `podsAfter`? I think it'll be helpful if this testcase failed.

Reply:
Oh man, nice catch!!! Thanks.

It actually takes a while before the evicted pod disappears from the node. Added `waitForTerminatingPodsToDisappear` check before checking the numbers. The test logs look like this now: